components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto.c

```diff
@@ -0,0 +1,184 @@
+    }
+
+    psa_close_key(handle);
+    return cose_ret;
+}
+
+enum t_cose_err_t
+t_cose_crypto_get_ec_pub_key(int32_t key_select,
+                             int32_t *cose_curve_id,
+                             struct useful_buf buf_to_hold_x_coord,
+                             struct useful_buf buf_to_hold_y_coord,
+                             struct useful_buf_c  *x_coord,
+                             struct useful_buf_c  *y_coord) {
+
+    enum tfm_plat_err_t err;
+    enum ecc_curve_t cose_curve;
+    struct ecc_key_t attest_key = {0};
+    uint8_t  key_buf[ECC_P_256_KEY_SIZE];
+
+    (void)key_select;
+
+    /* Get the initial attestation key */
+    err = tfm_plat_get_initial_attest_key(key_buf, sizeof(key_buf),
+                                          &attest_key, &cose_curve);
+
+    /* Check the availability of the private key */
+    if (err != TFM_PLAT_ERR_SUCCESS ||
+            attest_key.pubx_key == NULL ||
+            attest_key.puby_key == NULL)
+    {
+        return T_COSE_ERR_KEY_BUFFER_SIZE;
+    }
+
+    *cose_curve_id = (int32_t)cose_curve;
+
+    /* Check buffer size to avoid overflow */
+    if (buf_to_hold_x_coord.len < attest_key.pubx_key_size)
+    {
+        return T_COSE_ERR_KEY_BUFFER_SIZE;
+    }
+
+    /* Copy the X coordinate of the public key to the buffer */
+    memcpy(buf_to_hold_x_coord.ptr,
+           (const void *)attest_key.pubx_key,
+           attest_key.pubx_key_size);
+
+    /* Update size */
+    buf_to_hold_x_coord.len = attest_key.pubx_key_size;
+
+    /* Check buffer size to avoid overflow */
+    if (buf_to_hold_y_coord.len < attest_key.puby_key_size)
+    {
+        return T_COSE_ERR_KEY_BUFFER_SIZE;
+    }
+
+    /* Copy the Y coordinate of the public key to the buffer */
+    memcpy(buf_to_hold_y_coord.ptr,
+           (const void *)attest_key.puby_key,
+           attest_key.puby_key_size);
+
+    /* Update size */
+    buf_to_hold_y_coord.len = attest_key.puby_key_size;
+
+    x_coord->ptr = buf_to_hold_x_coord.ptr;
+    x_coord->len = buf_to_hold_x_coord.len;
+    y_coord->ptr = buf_to_hold_y_coord.ptr;
+    y_coord->len = buf_to_hold_y_coord.len;
+
+    return T_COSE_SUCCESS;
+}
+
+enum t_cose_err_t
+t_cose_crypto_hash_start(struct t_cose_crypto_hash *hash_ctx,
+                         int32_t cose_hash_alg_id) {
+    enum t_cose_err_t cose_ret = T_COSE_SUCCESS;
+    psa_status_t crypto_ret;
+
+    crypto_ret = psa_hash_setup(&hash_handle, PSA_ALG_SHA_256);
+
+    if (crypto_ret == PSA_ERROR_NOT_SUPPORTED)
+    {
+        cose_ret = T_COSE_ERR_UNSUPPORTED_HASH;
+    } else if (crypto_ret != PSA_SUCCESS)
+    {
+        cose_ret = T_COSE_ERR_HASH_GENERAL_FAIL;
+    }
+
+    return cose_ret;
+}
+
+void t_cose_crypto_hash_update(struct t_cose_crypto_hash *hash_ctx,
+                               struct useful_buf_c data_to_hash)
+{
+    if (data_to_hash.ptr != NULL) {
+        psa_hash_update(&hash_handle, data_to_hash.ptr, data_to_hash.len);
+    } else {
```

Comment:
why else needed?

Reply:
This function caled by get attestation token and get attestation token size. 
for the second purpose - data_to_hash(the payload) its null and no hash needed